src/components/EquippedCharacterSheetSlot/index.tsx

```diff
@@ -0,0 +1,106 @@
+import styled from "styled-components";
+import { AttributeImpact, CharacterSheetSlot, SlotType } from "../../types";
+import activeSpellPath from "./active_spell_icon.jpg";
+import instantSpellPath from "./instant_spell_icon.jpg";
+import weaponPath from "./weapon_icon.jpg";
+import Icon from "../Icon";
+import RenderedDice from "../RenderedDice";
+import { DiceAttribute } from "../../attributes";
+
+export interface Props {
+  slot: CharacterSheetSlot;
+}
+
+interface AttributeProps {
+  attributesImpacted: AttributeImpact[];
+}
+
+interface SlotIconsProps {
+  slotTypes: SlotType[];
+}
+
+const getIconPath = (powerType: SlotType) => {
+  if (powerType === "active") {
+    return activeSpellPath;
+  } else if (powerType === "instant") {
+    return instantSpellPath;
+  } else {
+    return weaponPath;
+  }
+};
+
+const EquippedCharacterSheetSlot = (props: Props) => {
+  return (
+    <Container>
+      <SlotIcons slotTypes={props.slot.slotTypes} />
+      <MainContent>
+        <Icon path={props.slot.iconLink} height={32} width={32} />
+        <CostBox>{props.slot.energyCost}</CostBox>
+        <div style={{ width: "100px" }}> {props.slot.name} </div>
+        <AttributesImpactedView attributesImpacted={props.slot.attributesImpacted} />
+      </MainContent>
+    </Container>
+  );
+};
+
+const AttributesImpactedView = (props: AttributeProps) => {
+  return (
+    <AttributesContainer>
+      {props.attributesImpacted
+        .filter((attr) => attr.attribute.name === "dice")
+        .sort()
+        .map((attr) => {
+          const dice = attr.attribute as DiceAttribute;
+          return <RenderedDice diceColour={dice.diceColour} numOfDice={attr.maxImpact} />;
+        })}
+    </AttributesContainer>
+  );
+};
+
+const SlotIcons = (props: SlotIconsProps) => {
+  return (
+    <SlotIconsContainer>
+      {props.slotTypes.map((slotType) => {
+        return <Icon height={18} width={18} path={getIconPath(slotType)} />;
```

Comment:
info: for cases where you only call a function to do some branching, we prefer a ternary statement to avoid the function call overhead.

you can ignore that for this line as i imagine we'll be exporting this from somewhere else for use elsewhere in the app.